brave/brave-core
# Use the infobar for WDP CTA instead of dialog

fix https://github.com/brave/brave-browser/issues/26576

Deleted previous WDP dialog.
Use Infobar UI for WDP CTA instead of modal dialog.
And removed bottom shadow from infobar container.

Light mode
<img width="1647" alt="Screen Shot 2022-11-22 at 2 29 34 PM" src="https://user-images.githubusercontent.com/6786187/203231020-d6f84a91-1d88-4cef-b62e-ff3d7396fe24.png">
Dark mode
<img width="1659" alt="Screen Shot 2022-11-22 at 2 30 00 PM" src="https://user-images.githubusercontent.com/6786187/203231030-c82aab8e-9368-420a-8c00-34d0fb8307b1.png">

This custom infobar has below three state based on width.

Wide layout with wide horizontal padding
<img width="1649" alt="Screen Shot 2022-11-22 at 2 31 36 PM" src="https://user-images.githubusercontent.com/6786187/203231579-f7e48b95-7949-4d85-82c1-c5259050d679.png">

Wide layout with narrow horizontal padding
<img width="816" alt="Screen Shot 2022-11-22 at 2 32 00 PM" src="https://user-images.githubusercontent.com/6786187/203231670-9566a9d5-99e9-44a3-b490-c1a1652c0629.png">

Narrow layout
<img width="572" alt="Screen Shot 2022-11-22 at 2 32 10 PM" src="https://user-images.githubusercontent.com/6786187/203231689-54838d00-9cf7-4165-8e4d-4d60b0ff1ec6.png">

![Screen_Recording_2022-11-22_at_2_33_37_PM_AdobeExpress](https://user-images.githubusercontent.com/6786187/203234014-574e7571-52af-4cbb-ae04-0b0e2c05e3d1.gif)



Resolves 

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

`npm run test brave_unit_tests -- --filter=WebDiscoveryCTATest*`
`npm run test brave_browser_tests -- --filter=WebDiscoveryTest*`

1. Launch brave with any profile(clean or existing)
2. Set non Brave Search as a default search engine
3. Load https://search.brave.com/ or with any query like https://search.brave.com/search?q=Brave&source=web
4. Check WDP infobar is not shown
5. Set Brave Search as a default search engine and enable WDP via brave://settings/?search=web+discovery
6. Repeat step 3 and check WDP infobar is not shown
7. disable WDP via brave://settings/?search=web+discovery
8. Repeat step3 and check WDP infobar is shown
9. Click x and check infobar is hidden
10. InfoBar is shown 5 times at most and once per day
11. To make infobar show again, edit `last_displayed` time like deleting last digit to simulate time passed more than 1 day from user dir's `Preferences` file after shutdown

```
      "web_discovery": {
         "cta_state": {
            "count": 2,
            "dismissed": true,
            "id": "v1",
            "last_displayed": "13313568560800197"
         }
      },
```
12. Relaunch Brave and check infobar is shown again when search.brave.com is loaded
13. Click `No thanks` and check infobar is not shown anymore
14. When new cta id is set, infobar will be shown although it's dimissed already
15. To simulate new cta id, change `id` in above Preferences to any other string
16. Check infobar is shown again when loading search.brave.com
